### ISSUE TITLE
Get block indices

### ISSUE THREAD

**Issue by jdmcclain47** [dask/dask#1643]
Hi, I would like to know if there is any way to get the block indices for an array.  Specifically, I want to chunk up an array, pass it through to some function by chunks, and then the function finds out the specific indices for the chunk and then produces output based on that.  For example, I may want something along the lines of:

```
import dask.array as da
from distributed import Client

totalsize = 10
blksize = 5
arr = da.ones((totalsize,totalsize),chunks=(blksize,blksize)

def get_max_x_index_in_block( inarr ):
     # since it is chunked in sizes of 5, the max index in the "x" 
     # direction will either be 4 or 9, depending on which block
     # was passed in
     return max_x_index

client = Client()
max_x = client.submit(get_max_x_index_in_block, arr)
max_x = max_x.result()
```

Now it may not be completely apparent why this would be necessary for an application, but it would actually be fairly useful for what I'm doing.  It seems simple enough but I couldn't find anything related to it.

**Comment by mrocklin** [comment]
See fairly low down in the API documentation for the `map_blocks` function: http://dask.pydata.org/en/latest/array-api.html#dask.array.core.map_blocks

In particular search for `block_id`.  It will give the function the block index rather than the array index, but you should hopefully be able to convert between the two with the `.chunks` attribute and a bit of elbow grease.

Also, you probably don't want to submit a function on a dask.array.  Instead use normal dask.array functions like map_blocks and then use `my_array.compute()` or `my_array = client.persist(my_array)`

**Comment by jdmcclain47** [comment]
Hmmm, I'm getting some weird errors here with the kwargs.  The first 2 print statements work, but the 3rd one fails.

```
import dask.array as da

nocc = 20
nvir = 4
nkpts = 8
nkpts_blksize = 2
t2   = da.ones((nkpts,nkpts,nkpts,nocc,nocc,nvir,nvir), \
         chunks=(nkpts_blksize,nkpts_blksize,nkpts_blksize,nocc,nocc,nvir,nvir))

def print_block_id1(arr1, block_id=None):
    print block_id
    return arr1

def print_block_id2(arr1, const):
    print const
    return arr1

def print_block_id3(arr1, const, block_id=None):
    print const
    print block_id
    return arr1

print "running print1..."
b = da.map_blocks(print_block_id1, t2, chunks=t2.chunks)
f = b.compute()

print "running print2..."
b = da.map_blocks(print_block_id2, t2, 8, chunks=t2.chunks)
f = b.compute()

print "running print3..."
b = da.map_blocks(print_block_id3, t2, 8, chunks=t2.chunks)
f = b.compute()
```

The error I'm getting is "TypeError: print_block_id3() got multiple values for keyword argument 'block_id'"

**Comment by jcrist** [comment]
The block_id issue was fixed in #1654, and is now merged into master. You can install the master branch easily by doing `pip install -U git+https://github.com/dask/dask.git`.
